### DISCUSSION TITLE
Copy new item to specific tab

### DISCUSSION THREAD

**Discussion by twopwood** [hluk/CopyQ#2145] [Q&A]
Sorry for such an elementary question but I'm trying to figure out to create a global shortcut command to copy an item directly to a specific tab.  Basically I want something similar to https://github.com/hluk/copyq-commands/blob/master/Automatic/image-tab.ini.  The closest I can get to is:

Attempt 1:
```
copyq:
tab('tmp')
copy()
insert(0, clipboard())
```
This creates an entry on both the main tab ('&clipboard') and the tab 'tmp'.  However this only copies the plain text.

Attempt 2:
```
copyq:
tab('tmp')
copy()
insert(0, input())
```
This copies all of the data (images, formatting, etc.) to '&clipboard' but just creates a blank entry in 'tmp'.

Ideally I'd like to copy all of the data just to the tab 'tmp'.

I feel like this is so simple that I'm missing some basic concept somewhere but I can't find a similar example and I can't figure out where to start in the documentation.  Thanks for any help pointing me in the right direction.

**Comment by twopwood** [comment]
OK, this gets at what I want to do but it works inconsistently and requires some fairly long `sleep` commands so I don't think this is the correct way.
```
    var tabName = 'tmp'
    var tabNameCurrent = config('clipboard_tab')
    config('clipboard_tab', tabName)
    sleep(600)
    copy()
    sleep(600)
    config('clipboard_tab', tabNameCurrent)
    sleep(200)
```
Does anyone have any other ideas?  I still think there must be a simple solution I'm missing...  Thanks.

**Comment by hluk** [comment]
One solution is to use a `settings()` to indicate a different output tab in **Global Shortcut** Command:

```js
const tab = 'tmp';
const maxWaitTimeMilliseconds = 2000;
const until = Date.now() + maxWaitTimeMilliseconds;
settings('use_clipboard_tab_until', [tab, until]);
copy();
```

and an **Automatic** command which checks the option and sets uses the output tab when new content appears:

```js
const [tab, until] = settings('clipboard_tab_until') || [];
if (tab && Date.now() < until) {
    setData(mimeOutputTab, tab);
    settings('use_clipboard_tab_until', []);
}
```
